bundles/net.tourbook/src/net/tourbook/ui/tourChart/ChartLayerNight.java

```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+ * Copyright (C) 2021 Frédéric Bard
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110, USA
+ *******************************************************************************/
+
+package net.tourbook.ui.tourChart;
+
+import net.tourbook.application.TourbookPlugin;
+import net.tourbook.chart.Chart;
+import net.tourbook.chart.GraphDrawingData;
+import net.tourbook.chart.IChartLayer;
+import net.tourbook.chart.IChartOverlay;
+import net.tourbook.preferences.ITourbookPreferences;
+
+import org.eclipse.jface.layout.PixelConverter;
+import org.eclipse.jface.preference.IPreferenceStore;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.graphics.GC;
+import org.eclipse.swt.graphics.Rectangle;
+
+public class ChartLayerNight implements IChartLayer, IChartOverlay {
+
+   private static final IPreferenceStore _prefStore = TourbookPlugin.getPrefStore();
+
+   private ChartNightConfig              _cnc;
+
+   public ChartLayerNight() {
+      //Nothing to do
+   }
+
+   /**
+    * This paints the night sections for the current graph configuration.
+    */
+   @Override
+   public void draw(final GC gc, final GraphDrawingData drawingData, final Chart chart, final PixelConverter pc) {
+
+      final int opacity = _prefStore.getInt(ITourbookPreferences.GRAPH_OPACITY_NIGHT_SECTIONS);
```

Comment:
I thought in a paint method it is very expensive to get pref values, but it can be ignored, just the first call is a little bit more expensive

![mt pref value code](https://user-images.githubusercontent.com/1283445/113724477-ae922f00-96f2-11eb-8b8f-d6bf11ed0beb.png)


![mt pref value](https://user-images.githubusercontent.com/1283445/113724475-adf99880-96f2-11eb-8e04-c60f25e2ac1d.png)